packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1477,32 +1491,46 @@ class DropdownButtonFormField<T> extends FormField<T> {
          autovalidate: autovalidate,
          builder: (FormFieldState<T> field) {
            final _DropdownButtonFormFieldState<T> state = field as _DropdownButtonFormFieldState<T>;
-           final InputDecoration effectiveDecoration = decoration.applyDefaults(
+           final InputDecoration decorationArg =  decoration ?? InputDecoration(focusColor: focusColor);
+           final InputDecoration effectiveDecoration = decorationArg.applyDefaults(
```

Comment:
I realize that we've been doing this all-along. I wish that InputDecorator computed the defaults.

Reply:
Yeah, and I wish I could access the value of `this.decoration` in a constructor, so I wouldn't have to compute the value twice.

Reply:
Although maybe I misunderstood your comment: did you want a change here?

Reply:
Just an observation, sorry about the confusion.